apollo-runtime/src/main/java/com/apollographql/apollo/internal/reader/RealResponseReader.java

```diff
@@ -203,7 +203,7 @@ Boolean readBoolean(Field field) throws IOException {
     }
   }
 
-  @SuppressWarnings("unchecked") private <T> T readCustomType(Field.CustomTypeField field) {
+  @SuppressWarnings("unchecked") private <T> T readCustomType(Field.CustomTypeField field) throws IOException {
```

Comment:
Yep, as an exception will now be thrown if the field does not exist in the recordSet.